client/solve.go

```diff
@@ -255,19 +282,49 @@ func (c *Client) solve(ctx context.Context, def *llb.Definition, runGateway runG
 			frontendInputs[key] = def.ToPB()
 		}
 
+		exports := make([]*controlapi.Exporter, 0, len(opt.Exports))
+		var localExporter *controlapi.Exporter
+		for _, exp := range exporters {
+			if exp.Type != ExporterLocal {
+				exports = append(exports, &controlapi.Exporter{
+					ID:    exp.id,
+					Type:  exp.Type,
+					Attrs: exp.Attrs,
+				})
+			} else if localExporter == nil {
+				// TODO(dima): different options per local exporter?
```

Comment:
I think if we do my suggestion below for multi-plexing between multiple output directories, this issue should go away? We then shouldn't need to do the de-duplication.

Reply:
I think I would need more context on this. It was a good pointer to avoid a new gRPC service - I did this with [metadata](https://github.com/moby/buildkit/pull/2760/files#diff-7078d905e2386549f621b6a42a2be0947615d2b0d51b1c0acb5f13caaf126ac6R273-R275) instead, but it does not remove the option for users to specify multiple outputs of type local (including options which may vary) - this was my confusion at this specific point.
Currently, only the first local export configuration block is used for all (if >1 specified on command line) and then the destination directories from each corresponding local export are used to duplicate the results of the first export. Sorry if I'm misunderstanding your concern and suggestion - would appreciate more feedback.